application/FtbListModel.cpp

```diff
@@ -117,3 +152,45 @@ FtbModpack FtbListModel::at(int row)
 {
 	return modpacks.at(row);
 }
+
+void FtbListModel::logoLoaded(QString logo, QPixmap out)
+{
+	m_loadingLogos.removeAll(logo);
+	m_logoMap.insert(logo, out);
+	emit dataChanged(createIndex(0, 0), createIndex(modpacks.size(), 0));
+}
+
+void FtbListModel::logoFailed(QString logo)
+{
+	m_failedLogos.append(logo);
+	m_loadingLogos.removeAll(logo);
+}
+
+void FtbListModel::requestLogo(QString file)
+{
+	if(m_loadingLogos.contains(file) || m_failedLogos.contains(file)) {
+		return;
+	}
+
+	MetaEntryPtr entry = ENV.metacache()->resolveEntry("FTBPacks", QString("logos/%1").arg(file.section(".", 0, 0)));
+	NetJob *job = new NetJob(QString("FTB Icon Download for %1").arg(file));
+	job->addNetAction(Net::Download::makeCached(QUrl(QString("https://ftb.cursecdn.com/FTB2/static/%1").arg(file)), entry));
+
+	auto fullPath = entry->getFullPath();
+	QObject::connect(job, &NetJob::finished, this, [this, file, fullPath]{
+		QPixmap pixmap;
+		pixmap.load(fullPath);
+		pixmap = pixmap.scaled(QSize(42, 42));
```

Comment:
`// FIXME: rewrite to use QIcon, avoid resizing the images, set icon size in the view`